### ISSUE TITLE
Compile on Windows error: Reference Xamarin.Mac not resolved

### ISSUE THREAD

**Issue by wangkaibule** [mono/monodevelop#9099]
I'm trying to build Monodevelop on [Appveyor](appveyor.com) with Visual Studio 2019 environment, but following error has occurred: 

```
[00:05:03]        "C:\projects\monodevelop\monodevelop\main\Main.sln" (default target) (1:2) ->
[00:05:03]        "C:\projects\monodevelop\monodevelop\main\src\core\MonoDevelop.Ide\MonoDevelop.Ide.csproj" (default target) (13:59) ->
[00:05:03]        (CoreCompile target) -> 
[00:05:03]          MonoDevelop.Ide.Fonts\FontChooserPanelWidget.cs(32,7): error CS0246: The type or namespace name 'AppKit' could not be found (are you missing a using directive or an assembly reference?) [C:\projects\monodevelop\monodevelop\main\src\core\MonoDevelop.Ide\MonoDevelop.Ide.csproj]
[00:05:03]          MonoDevelop.Ide.Fonts\FontChooserPanelWidget.cs(33,7): error CS0246: The type or namespace name 'Foundation' could not be found (are you missing a using directive or an assembly reference?) [C:\projects\monodevelop\monodevelop\main\src\core\MonoDevelop.Ide\MonoDevelop.Ide.csproj]
[00:05:03] 
[00:05:03] 
[00:05:03]        "C:\projects\monodevelop\monodevelop\main\Main.sln" (default target) (1:2) ->
[00:05:03]        "C:\projects\monodevelop\monodevelop\main\tests\Ide.Tests\MonoDevelop.Ide.Tests.csproj" (default target) (63:8) ->
[00:05:03]        "C:\projects\monodevelop\monodevelop\main\tests\MonoDevelop.Core.Tests\MonoDevelop.Core.Tests.csproj" (default target) (80:8) ->
[00:05:03]          MonoDevelop.Projects\ProjectTests.cs(1016,54): error CS0234: The type or namespace name 'Composition' does not exist in the namespace 'System.ComponentModel' (are you missing an assembly reference?) [C:\projects\monodevelop\monodevelop\main\tests\MonoDevelop.Core.Tests\MonoDevelop.Core.Tests.csproj]
```
It seems like the Xamarin.Mac reference cannot be resolved.
The full log is attached below and the Appveyor repository is [here](https://ci.appveyor.com/project/wangkaibule/monodevelop)
[log (1).txt](https://github.com/mono/monodevelop/files/3774342/log.1.txt)

**Comment by Therzok** [comment]
Not sure about core tests though, it should be picked up by [CoreReferences](https://github.com/mono/monodevelop/blob/xammac-font/main/msbuild/CoreReferences.props)

**Comment by Therzok** [comment]
Huh, interesting: https://github.com/mono/monodevelop/blob/152bf0867f774e2fdaa10adac063ccb1bbd31ca2/main/msbuild/MonoDevelop.BeforeCommon.targets#L95

It seems like they're only added if that prop is set to true. But given some other test projects reference them, I'll just add a reference in the csproj.

**Comment by Therzok** [comment]
@wangkaibule can you trigger an appveyor build tageting [this branch](https://github.com/mono/monodevelop/tree/xammac-font) ?

**Comment by wangkaibule** [comment]
Sorry for the delay, the commit seems worked but more problems surfaced from other project. Please refer to the log file attached on this comment.
[SecondRound.txt](https://github.com/mono/monodevelop/files/3774477/SecondRound.txt)

**Comment by wangkaibule** [comment]
And to whomever interested, this is the configuration I'm using on Appveyor:

``` yaml
version: 1.0.{build}
branches:
  only:
  - xammac-font
image: Visual Studio 2019
init:
- ps: >-
    Invoke-WebRequest https://xamarin.azureedge.net/GTKforWindows/Windows/gtk-sharp-2.12.45.msi -outfile gtk-sharp-2.12.45.msi

    Invoke-WebRequest https://files.xamarin.com/~jeremie/MonoLibraries.msi -outfile MonoLibraries.msi
clone_script:
- ps: git clone git://github.com/mono/monodevelop -bxammac-font -j8 --recursive --single-branch -q
install:
- cmd: >-
    msiexec /i MonoLibraries.msi /passive /le loa.log

    msiexec /i gtk-sharp-2.12.45.msi /passive /le lob.log


    type loa.log

    type lob.log
build_script:
- cmd: >-
    cd monodevelop

    cd main

    winbuild.bat
```

**Comment by Therzok** [comment]
Yeah, seems like something is broken with a commit I made. See https://github.com/mono/monodevelop/pull/9097

**Comment by knocte** [comment]
@Therzok hey, if I contributed a GithubActionsCI workflow (as I guess AppVeyor is less MS-friendly ;) ) that runs compilation on Windows, would that PR be accepted? so that we catch these breakages early on

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.6.0.208 (master)

Pull Request #9100 merged by: Jose Miguel Torres
Author: mono
Commit: c740bfab1077385701ebf3486970fef4d9e44a86 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.5.0.1066 (release-8.5)

Pull Request #9100 merged by: Jose Miguel Torres
Author: mono
Commit: c740bfab1077385701ebf3486970fef4d9e44a86 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.6.0.802 (master)

Pull Request #9100 merged by: Jose Miguel Torres
Author: mono
Commit: c740bfab1077385701ebf3486970fef4d9e44a86 (xamarin/monodevelop)
